lib/media-controls-impl.js

```diff
@@ -0,0 +1,524 @@
+  this.mediaPlayer.addEventListener('seeked', this);
+  this.mediaPlayer.addEventListener('ended', this);
+};
+
+MediaControlsImpl.prototype.removeEventListeners = function() {
+  this.shadowRoot.removeEventListener('touchstart', this);
+  this.shadowRoot.removeEventListener('touchmove', this);
+  this.shadowRoot.removeEventListener('touchend', this);
+  this.shadowRoot.removeEventListener('mousedown', this);
+
+  this.mediaPlayer.removeEventListener('loadedmetadata', this);
+  this.mediaPlayer.removeEventListener('play', this);
+  this.mediaPlayer.removeEventListener('pause', this);
+  this.mediaPlayer.removeEventListener('timeupdate', this)
+  this.mediaPlayer.removeEventListener('seeked', this);
+  this.mediaPlayer.removeEventListener('ended', this);
+
+  if (this.mouseEventHandlerRegistered) {
+    this.shadowRoot.removeEventListener('mousemove', this, true);
+    this.shadowRoot.removeEventListener('mouseup', this, true);
+  }
+
+  this.mouseEventHandlerRegistered = false;
+};
+
+MediaControlsImpl.prototype.handleEvent = function(e) {
+
+  // If we get a touchstart, don't process mouse events
+  if (e.type === 'touchstart') {
+    this.shadowRoot.removeEventListener('mousedown', this);
+    this.shadowRoot.removeEventListener('mousemove', this);
+    this.shadowRoot.removeEventListener('mouseup', this);
+  }
+
+  switch(e.type) {
+
+    case 'mousedown':
+      //
+      // The component is listening to window 'mousemove' events so
+      // that the slider movement will function even when the mouse
+      // moves off the play head. However, if the component is always
+      // listening to the window events, it would receive very many
+      // spurious 'mousemove' events. To prevent this, the component
+      // only listens to 'mousemove' events after receiving a 'mousedown'
+      // event.
+      //
+      if (!this.mouseEventHandlerRegistered) {
+        window.addEventListener('mousemove', this, true);
+        window.addEventListener('mouseup', this, true);
+        this.mouseEventHandlerRegistered = true;
+      }
+
+      // fall through to touchstart...
+
+    case 'touchstart':
+      switch(e.target) {
+        case this.els.play:
+          //
+          // Let the 'play' and 'pause' handlers take care of changing
+          // the icon and setting the l10n-id (for the screen reader).
+          //
+          if (this.mediaPlayer.paused) {
+            this.mediaPlayer.play();
+          }
+          else {
+            this.mediaPlayer.pause();
+          }
+          break;
+
+        case this.els.seekForward:
+        case this.els.seekBackward:
+
+          var direction = null;
+          if (e.target === this.els.seekForward) {
+            direction = 1;
+          } else if (e.target === this.els.seekBackward) {
+            direction = -1;
+          } else {
+            return;
+          }
+
+          var offset = direction * 10;
+          this.seekBy(this.mediaPlayer.currentTime + offset);
```

Comment:
I see I was wrong above. All three buttons respond on touchstart.  Two comments here:

1) the name seekBy() strongly implies a relative amount, like you're seeing by + or - 10 seconds. But you're passing an absolute time here, so the function should be renamed to seekTo().

2) For the music app, these forward and back buttons are used for "next song" and "previous song" if you just tap them, and are used to seek only when held down.  So if we're going to use this component more generally, we'll want changes here.  That shouldn't stop you from getting this initial commit landed, however.